From 2c9609b3c1a12ee9e6ed9dca9b6b7aeb0d06e415 Mon Sep 17 00:00:00 2001
From: "Guoniu.zhou" <guoniu.zhou@nxp.com>
Date: Wed, 20 Jul 2022 20:54:28 +0800
Subject: [PATCH 252/274] LF-6635: media: i2c: fix kernel cash issue when no
 ap1302.fw in rootfs

Fix kernel cash issue when no ap1302.fw in rootfs

Signed-off-by: Guoniu.zhou <guoniu.zhou@nxp.com>
---
 drivers/media/i2c/ap1302.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/media/i2c/ap1302.c b/drivers/media/i2c/ap1302.c
index 611fd6108..c037e282e 100644
--- a/drivers/media/i2c/ap1302.c
+++ b/drivers/media/i2c/ap1302.c
@@ -236,13 +236,8 @@ static void ap1302_fw_handler(const struct firmware *fw, void *context)
 	int ret;
 
 	if (fw == NULL) {
-		ret = request_firmware_nowait(THIS_MODULE, FW_ACTION_UEVENT,
-					"imx/camera/ap1302.fw", dev, GFP_KERNEL,
-					ap1302_dev, ap1302_fw_handler);
-		if (ret) {
-			dev_err(dev, "Failed request_firmware err %d\n", ret);
-			return;
-		}
+		dev_err(dev, "Failed to request_firmware\n");
+		return;
 	}
 
 	ap1302_fw = (struct ap1302_firmware *)fw->data;
-- 
2.25.1

